src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -403,30 +559,103 @@ public SymbolicMatch FindMatch(RegexRunnerMode mode, ReadOnlySpan<char> input, i
             }
             else
             {
-                Registers endRegisters = _pattern._info.ContainsLineAnchor ?
+                Registers endRegisters = _containsAnyAnchor ?
                     FindSubcaptures<FullInputReader>(input, matchStart, matchEnd, perThreadData) :
                     FindSubcaptures<NoZAnchorInputReader>(input, matchStart, matchEnd, perThreadData);
                 return new SymbolicMatch(matchStart, matchEnd - matchStart, endRegisters.CaptureStarts, endRegisters.CaptureEnds);
             }
         }
 
+        /// <summary>
+        /// This version of <see cref="FindEndPositionFallback"/> uses a different set of interfaces,
```

Comment:
And there's no good way to just share a single set of interfaces even if they have more surface area?

Reply:
Elaborate on this, do you mean adding more arguments to the interface? The idea here was to start from a clean slate and with most scenarios that i know will work. The other one i would leave as-is because there was complex logic and unnecessary variables/operations in the other function which i didn't want to consider in the process.